geomesa-hbase/geomesa-hbase-datastore/src/main/scala/org/locationtech/geomesa/hbase/data/HBaseIndexAdapter.scala

```diff
@@ -192,40 +194,35 @@ class HBaseIndexAdapter(ds: HBaseDataStore) extends IndexAdapter[HBaseDataStore]
     // index api defines start row inclusive, end row exclusive
     // both these conventions match the conventions for hbase scan objects
     val ranges = byteRanges.map {
-      case BoundedByteRange(start, end) => new Scan(start, end)
-      case SingleRowByteRange(row)      => new Scan(row, ByteArrays.rowFollowingRow(row)).setSmall(true)
+      case BoundedByteRange(start, stop) => new RowRange(start, true, stop, false)
+      case SingleRowByteRange(row)       => new RowRange(row, true, ByteArrays.rowFollowingRow(row), false)
     }
+    val small = byteRanges.headOption.exists(_.isInstanceOf[SingleRowByteRange])
 
     val tables = index.getTablesForQuery(filter.filter).map(TableName.valueOf)
     val (colFamily, schema) = groups.group(index.sft, hints.getTransformDefinition, ecql)
 
     val transform: Option[(String, SimpleFeatureType)] = hints.getTransform
 
+    def empty(reducer: Option[FeatureReducer]): Option[HBaseQueryPlan] =
```

Comment:
*blinks* Ok, what does this function do?!

Reply:
it checks for and creates an empty scan plan, but with the appropriate reduce step that's needed for e.g. arrow or stat queries.